src/main/java/org/jenkinsci/plugins/builduser/BuildUserStep.java

```diff
@@ -0,0 +1,76 @@
+package org.jenkinsci.plugins.builduser;
+
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.HashSet;
+import java.util.Map;
+import java.util.Set;
+
+import org.jenkinsci.plugins.builduser.varsetter.IUsernameSettable;
+import org.jenkinsci.plugins.workflow.steps.Step;
+import org.jenkinsci.plugins.workflow.steps.StepContext;
+import org.jenkinsci.plugins.workflow.steps.StepDescriptor;
+import org.jenkinsci.plugins.workflow.steps.StepExecution;
+import org.kohsuke.stapler.DataBoundConstructor;
+
+import hudson.Extension;
+import hudson.model.Run;
+
+/**
+ * Get and return build user variables, see {@link IUsernameSettable}:
+ *
+ * @author awitt
+ * @see IUsernameSettable
+ */
+public class BuildUserStep extends Step {
+
+    @DataBoundConstructor
+    public BuildUserStep() {
+    }
+
+    @Override
+    public StepExecution start(StepContext context) {
+        return new Execution(context);
+    }
+
+    public static class Execution extends StepExecution {
+
+        private static final long serialVersionUID = 1L;
+
+        public Execution(StepContext context) {
+            super(context);
+        }
+
+        @Override
+        public boolean start() throws Exception {
+            Run<?, ?> build = getContext().get(Run.class);
+            Map<String, String> variables = new HashMap<>();
```

Comment:
By definition, a Map returns null when the key is non existent (see [`get(Object key)`](https://docs.oracle.com/javase/8/docs/api/java/util/Map.html#get-java.lang.Object-). This means that the users who make a typo will have silent failures, e.g. `getBuildUser().BULD_USER_ID` will return the string `null`.

I'd prefer an outright failure.

Reply:
@martinda I understand your concern and agree that an outright failure would be a way to smooth out errors such as typos. But, after thinking about it for a while, I'm not sure how it can be done in this case. Do you have any ideas or suggestions?
Thanks!

Reply:
@jenkinsci/code-reviewers can someone provide a suggestion/approach to overcome the issue presented above?